spectra-generator/src/main/java/se/de/hu_berlin/informatik/gen/spectra/predicates/pdg/CodeLocation.java

```diff
@@ -0,0 +1,39 @@
+package se.de.hu_berlin.informatik.gen.spectra.predicates.pdg;
+
+import soot.SootMethod;
+import soot.Unit;
+
+public class CodeLocation {
+
+    public Unit unit;
+    public String className;
+    public String methodName;
+    public SootMethod method;
+    public SootConnector sootConnector;
```

Comment:
Is all of this really needed later? Every reference to objects prevent the garbage collector to throw these objects away. So idk if you really need access to the unit, the soot method or the connector later. Maybe it's enough to store the line number (since you access it in the getLocationString method).

Reply:
yes its is needed. 
I could store the method name instead of the full method but that just leads to Soot parsing the Name and then loading the Method again.
the `SootConnector` is for `target.sootConnector.getIteratorOnCallsOutOfMethod(target.method);` in GenCodeLocationBasedRanking Line 254

Reply:
the garbage collection is a moot point as all `SootConnectors` are items in a static Map for caching reasons.